### DISCUSSION TITLE
If I pick a region in Europe for serverless functions, am I guaranteed they only run in that region?

### DISCUSSION THREAD

**Discussion by MihaiAnton** [vercel/community#2314] [Help]
### Summary

I am building an app that processes data for users in the EU, thus it has to be GDPR compliant. As per the docs at https://vercel.com/docs/concepts/functions/serverless-functions/regions, I should be able to pick Frankfurt, for example, as a region for my functions. 

Does Vercel treat this as a preference, and will try to assign my functions to the region, and in case of failure, pick another region (even outside the EU), or would this be treated as a requirement, and all the functions will strictly run in the chosen region?

**Comment by smohadjer** [comment]
Note that even if you do so, the rest of your app still won't run in EU, only your serverless functions will do so.

**Reply by jamesarosen** [reply]
Can you elaborate on what you mean by "the rest of your app"? Do you mean Vercel services other than serverless functions? (e.g. edge functions, config stores, blob, KV, postgres)

**Reply by smohadjer** [reply]
I meant any static page in your app or website.
